### PR TITLE
Refactor `indent` handling in asdl_cpp.py

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Closes: #405
Closes: https://gitlab.com/lfortran/lfortran/-/issues/533

### CODE REVIEW

File: src/libasr/asdl_cpp.py [hunk 1]
```diff
@@ -1190,12 +1190,10 @@ def visitModule(self, mod):
...
         self.emit("public:")
-        self.emit(  "std::string s, indtd;", 1)
+        self.emit(  "std::string s, indtd = \"\";", 1)
         self.emit(  "bool use_colors;", 1)
-        self.emit(  "bool indent, start_line = true;", 1)
-        self.emit(  "int indent_level = 0, indent_spaces = 3, lvl = 0;", 1)
-        self.emit(  "int tmp = 0, tmp1 = 0, tmp2 = 2;", 1)
-        self.emit(  "int curly[2000], round[2000];", 1)
+        self.emit(  "bool indent;", 1)
+        self.emit(  "int indent_level = 0, indent_spaces = 4;", 1)
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/asdl_cpp.py:1196]: I increased the indent_space here, but I wonder what size would be preferable?

**Comment by Thirumalai-Shaktivel** [comment]
I just now saw this message:
```
# ASR output
#   * Each s-expression of the form (HEAD args) represents a node from grammar/ASR.asdl
#   * Consult ASR.asdl for the types and number of arguments
#   * Faithful representation (no information omitted)
#   * You can turn off this comment by `--no-asr-comment`
# Abbreviations made:
# (Int4)         -> (Integer 4 [])
# 123            -> (ConstantInteger 123 (Integer 4 []))
# (Int4 [(1 4)]) -> (Integer 4 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])
# (* A B C D)    -> (BinOp A Mul B C D)
# (= A B C D)    -> (Assignment A B C D)
#
(TranslationUnit
(SymbolTable 1 {
                                                
expr2: (Program                                                       
(SymbolTable 2 {                                                      
  x: (Variable 2 x Local () () Default (Int4)
      Source Public Required .false.)
  y: (...
      ...)
})                                                                    
    expr2 []    
[
  (= (Var 2 x) (* (+ 2 3 (Int4) 5) 5) (Int4) 25)            
  (= (Var 2 x) (* 2 (Var 2 x) (Int4)))                         
  (Print () [(Var 2 x)])
]      
)     

}) [])
```
I will refactor indent to look like this, apart from this any other changes to be made, please let me know.

**Comment by certik** [comment]
Thanks for working on this. Does this fix the `--indent` issue?

I noticed that this PR's do loop formatting:
```
                (DoLoop
                    0
                    ()
                    0
                    i
                    1
                    10
                    ()
                    [
                        (=
                            0
                            x
                            (* i x)
                            ()
                        )
                    ]
                    ()
                    ()
                )
```
seems less readable than in master:
```
      (DoLoop 0 () 0 i 1 10 () [
         (= 0 x (* i x) ())] () ())
```

If we are doing this refactoring, then let's do it according to the comment you posted, I think that will be very readable.

**Comment by Thirumalai-Shaktivel** [comment]
> Thanks for working on this. Does this fix the --indent issue?
Yup, this fixes the indent issue!

**Comment by certik** [comment]
Keep this PR open. Let's see if we can get the output we want, you can send it to another PR. If it is too complicated, we can then merge this PR to at least get a fix in.

**Comment by Thirumalai-Shaktivel** [comment]
ASR looks neat and clean, but AST!
@certik This is ready!

**Review by certik — APPROVED** [review]
I think this is fine. If it fixes the "assert" issue, then let's use this.

We can later improve how it looks. 

Before you merge, make sure you rename the commit (WIP) to something more descriptive, or use the squash button.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the review!

**Comment by Shaikh-Ubaid** [comment]
Thank you so much for this @Thirumalai-Shaktivel.

The `ASR` is now visible on `dev.lfortran.org` for the `mandelbrot` example.
image

Also, @certik Thank you for setting the `SSH Keys`, the `CI` seems to work.

**Comment by certik** [comment]
Here is the automatic commit: https://github.com/lfortran/lcompilers_frontend/commit/8deb3daf3c4d35b345c0784a837472a340abb816

**Pull request merged**